test/data/deploy/inference-service-with-model-version.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
```

Comment:
I don't see a test where `mr-registered-model-id` is absent and `mr-model-version-id` is present.
If this condition is not valid, the reconciler should properly validate it and there should be a test.

Reply:
Yeah, the condition you mentioned is not valid - refactoring a bit the code to make it much more explicit and adding a test

Reply:
Added a test covering this case as well